oxygen-account/src/test/java/com/oxygenxml/account/user/UserControllerTest.java

```diff
@@ -36,117 +38,139 @@
+		UserDto newUserDto = new UserDto();
+		newUserDto.setName("Test");
+		newUserDto.setEmail("invalidemail");
+		newUserDto.setPassword("password");
+
+		ResultActions resultAction = mockMvc.perform(post("/api/users/register")
+				.contentType("application/json")
+				.content(JsonUtil.asJsonString(newUserDto)));
+
+		resultAction.andExpect(status().isUnprocessableEntity())
+		.andExpect(jsonPath("$.errors[0].errorMessage", is(Message.EMAIL_INVALID.getMessage())))
+		.andExpect(jsonPath("$.internalErrorCode", is(1008)))
+		.andExpect(jsonPath("$.errors[0].messageId", is(Message.EMAIL_INVALID.getId())));
+
+	}
+
+	/**
+	 * The testInvalidPassword tests the registration with an invalid password.
+	 * @throws Exception
+	 */
+	@Test
+	void testInvalidPassword() throws Exception{
+		UserDto newUserDto = new UserDto();
+		newUserDto.setName("Test");
+		newUserDto.setEmail("email@email.com");
+		newUserDto.setPassword("pass");
+
+		ResultActions resultAction = mockMvc.perform(post("/api/users/register")
+				.contentType("application/json")
+				.content(JsonUtil.asJsonString(newUserDto)));
+
+		resultAction.andExpect(status().isUnprocessableEntity())
+		.andExpect(jsonPath("$.errors[0].errorMessage", is(Message.SHORT_FIELD.getMessage())))
+		.andExpect(jsonPath("$.internalErrorCode", is(1008)))
+		.andExpect(jsonPath("$.errors[0].messageId", is(Message.SHORT_FIELD.getId())));
+
+	}
+
+	/**
+	 * The testEmptyField tests the registration with an invalid password.
+	 * @throws Exception
+	 */
+	@Test
+	void testEmptyField() throws Exception{
+		UserDto newUserDto = new UserDto();
+		newUserDto.setName("");
+		newUserDto.setEmail("email@email.com");
+		newUserDto.setPassword("password");
+
+		ResultActions resultAction = mockMvc.perform(post("/api/users/register")
+				.contentType("application/json")
+				.content(JsonUtil.asJsonString(newUserDto)));
+
+		resultAction.andExpect(status().isUnprocessableEntity())
+		.andExpect(jsonPath("$.errors[0].errorMessage", is(Message.EMPTY_FIELD.getMessage())))
+		.andExpect(jsonPath("$.internalErrorCode", is(1008)))
+		.andExpect(jsonPath("$.errors[0].messageId", is(Message.EMPTY_FIELD.getId())));
+
+	}
+
+	/**
+	 * The testMultipleErrors tests that the MultipleOxygenAccountException is throwen correctly
+	 * @throws Exception
+	 */
+	@Test
+	void testMultipleErrors() throws Exception{
+		UserDto newUserDto = new UserDto();
+		newUserDto.setName("");
+		newUserDto.setEmail("emailemail.com");
+		newUserDto.setPassword("pass");
+
+		ResultActions resultAction = mockMvc.perform(post("/api/users/register")
+				.contentType("application/json")
+				.content(JsonUtil.asJsonString(newUserDto)));
+
+		resultAction.andExpect(status().isUnprocessableEntity())
+		.andExpect(jsonPath("$.internalErrorCode", is(1008)))
+		.andExpect(jsonPath("$.errors[*].errorMessage", containsInAnyOrder(Message.EMPTY_FIELD.getMessage(), Message.SHORT_FIELD.getMessage(), Message.EMAIL_INVALID.getMessage())))
```

Comment:
1. Nu este verificat fieldName-ul din erori.
2. Verificarea erorilor ar trebui sa fie individuala, adica verificam informatiile (message, messageId, fieldName) pentru fiecare eroare in parte. De exemplu, se poate rescrie o parte din asserturi astfel:
```
resultActions.andExpect(jsonPath("$.errors[?(@.fieldName == 'name')].messageId", hasItem(Message.EMPTY_FIELD.getId())));
resultActions.andExpect(jsonPath("$.errors[?(@.fieldName == 'name')].errorMessage", hasItem(Message.EMPTY_FIELD.getMessage())));
```
sau se poate aserta astfel:
```
resultActions.andExpect(jsonPath("$.errors.[0].fieldName", is("name")));
resultActions.andExpect(jsonPath("$.errors.[0].messageId", is(Message.EMPTY_FIELD.getId())));
resultActions.andExpect(jsonPath("$.errors.[0].errorMessage", is(Message.EMPTY_FIELD.getMessage())));
```
La fel se continua cu asserturile si pentru celelalte fielduri care au erori.